lib/src/parser/recursive.dart

```diff
@@ -307,41 +328,139 @@ class RecursiveAstParser {
+          _consumeWhitespaces();
+          equalSign = _reader.next();
+          _consumeWhitespaces();
+          valueToken = _reader.next();
+        }
       }
-      _reader.expect(NgTokenType.beforeElementDecoratorValue);
-      valueToken = _reader.expect(NgTokenType.elementDecoratorValue);
-      _reader.expect(NgTokenType.openElementEnd);
-    } else if (nextToken.type != NgTokenType.openElementEnd) {
-      _reader.error('Expected ">", got $nextToken');
-      return null;
     }
-    _reader.expect(NgTokenType.closeElementStart);
-    final closeName = _reader.expect(NgTokenType.elementIdentifier);
-    if (closeName.lexeme != 'ng-content') {
-      _reader.error(''
-          'Expected closing tag for "ng-content", got '
-          '"${closeName.lexeme}"');
-      return null;
+
+    _consumeWhitespaces();
+
+    // Ensure closed by '>' and not '/>'.
+    endToken = _reader.next();
+    if (endToken.type == NgTokenType.openElementEndVoid) {
+      var e = new AngularParserException(
+        '"ng-content" is not a valid void element',
+        endToken.lexeme,
+        endToken.offset,
+      );
+      exceptionHandler.handle(e);
+      endToken = new NgToken.generateErrorSynthetic(
+        endToken.offset,
+        NgTokenType.openElementEnd,
+      );
+    }
+
+    // Ensure closing </ng-content> exists.
+    if (_reader.peekType() != NgTokenType.closeElementStart) {
+      var e = new AngularParserException(
+        'Expected closing tag for "ng-content".',
+        _reader.peek().lexeme,
+        _reader.peek().offset,
+      );
+      exceptionHandler.handle(e);
+      closeElementAst = new CloseElementAst('ng-content');
+    } else {
+      var closeElementStart = _reader.next();
+      var closeElementName = _reader.peek().lexeme;
+      var closeElementOffset = _reader.peek().offset;
+
+      if (closeElementName != 'ng-content') {
+        var e = new AngularParserException(
+          'Expected closing tag to match "ng-content", instead got $closeElementName',
+          closeElementName,
+          closeElementOffset,
+        );
+        exceptionHandler.handle(e);
+        _reader.putBack(closeElementStart);
+        closeElementAst = new CloseElementAst('ng-content');
+      } else {
+        closeElementAst = parseCloseElement(closeElementStart);
+      }
     }
-    final endToken = _reader.expect(NgTokenType.closeElementEnd);
     return new EmbeddedContentAst.parsed(
       _source,
       beginToken,
+      elementIdentifierToken,
       endToken,
-      valueToken?.innerValue,
+      closeElementAst,
+      selectToken,
+      equalSign,
+      valueToken,
     );
   }
 
+  /// Helper function that accumulates all parts of attribute-value variant
+  /// and returns it as a single string. Should be used to gather any
+  /// non-'select' decorator. Consumes all necessary erroneous tokens.
+  String _accumulateInvalidNgContentDecoratorValue(
+      NgToken beforeElementDecorator, NgToken nextToken) {
```

Comment:
What about in the case of <ng-content blah="value" select="something" [prop]="value3">, this would mean that the the entire attribute name-value chains would be removed. Ideally I would want this to still keep the 'select="something"'.